cabal-install/Distribution/Client/CmdRun.hs

```diff
@@ -90,7 +121,8 @@ runAction (configFlags, configExFlags, installFlags, haddockFlags)
     baseCtx <- establishProjectBaseContext verbosity cliConfig
 
     targetSelectors <- either (reportTargetSelectorProblems verbosity) return
-                   =<< readTargetSelectors (localPackages baseCtx) targetStrings
+                   =<< readTargetSelectors (localPackages baseCtx)
```

Comment:
It is too bad there is no singular version of readTargetSelectors, since it seems like that is what you want.